Projects/19-Captsone_01/ispy1/predictive_statistics.py

```diff
@@ -0,0 +1,255 @@
+from sklearn import preprocessing
+from sklearn import metrics
+from sklearn.model_selection import train_test_split, GridSearchCV
+from sklearn import metrics, linear_model
+from imblearn import over_sampling
+from sklearn.calibration import CalibratedClassifierCV
+from sklearn.ensemble import RandomForestClassifier as RFC
+import numpy as np
+import matplotlib.pyplot as plt
+
+RANDOM_STATE = 42; # for reproducibility
+
+
+def labels_to_numbers(DataFrame, Variable):
+    le = preprocessing.LabelEncoder()
+    numbers_ = le.fit_transform(DataFrame[Variable].values)
+    return numbers_
+
+
+def binary_classifier_metrics(classifier, Xtrain, Ytrain, Xtest, Ytest):
+    # metrics
+    predicted_class = classifier.predict(Xtest)
+
+    kappa = metrics.cohen_kappa_score(Ytest,predicted_class)
+    np.round(kappa,3)
+
+    auc = metrics.roc_auc_score(Ytest,predicted_class)
+    auc = np.round(auc,3)
+
+
+    # ROC curve
+    probability = classifier.predict_proba(Xtest)
+    fpr, tpr, _  = metrics.roc_curve(Ytest, probability[:,1], drop_intermediate=False)
+
+    # report
+    print(metrics.classification_report(Ytest,predicted_class))
+    print('The estimated Cohen kappa is ' + str(kappa))
+    print('The estimated AUC is ' + str(auc))
```

Comment:
Awesome !!!
I had been wondering how to use % and .format() properly. 
Then, I am assuming that I should do something like the following for lines 37 and 38:

```Python
In [1]: kappa = 0.43123434
In [2]: '{} {:3.2f}'.format('Cohen kappa  is:', kappa)
Out[2]: 'Cohen kappa  is: 0.43'

```